src/utils/function.js

```diff
@@ -44,11 +44,50 @@ export function rateLimit(win, callback, minInterval) {
     }
   }
 
-  return function() {
+  return function(...args) {
     if (locker) {
-      nextCallArgs = arguments;
+      nextCallArgs = args;
     } else {
-      fire(arguments);
+      fire(args);
+    }
+  };
+}
+
+/**
+ * Wraps a given callback and applies a wait timer, so that minInterval
+ * milliseconds must pass since the last call before the callback is actually
+ * invoked.
+ *
+ * @param {!Window} win
+ * @param {function()} callback
+ * @param {number} minInterval the minimum time interval in millisecond
+ * @returns {function()}
+ */
+export function debounce(win, callback, minInterval) {
+  let locker = 0;
+  let timestamp = 0;
+  let nextCallArgs = null;
+
+  function fire(args) {
+    nextCallArgs = null;
+    callback.apply(null, args);
+  }
+
+  function waiter() {
+    locker = 0;
+    const remaining = minInterval - (Date.now() - timestamp);
+    if (remaining > 0) {
+      locker = win.setTimeout(waiter, remaining);
+    } else {
+      fire(nextCallArgs);
+    }
+  }
+
+  return function(...args) {
+    timestamp = Date.now();
```

Comment:
documentation wasn't clear to me (`since the last call` didn't register to me to imply `of course also the first call since there is no last call` :) ) I think first call is special enough to deserve its own sentence.

`PositionObserver` definitely should act on the first `onScroll` event it receives immediately.